src/kernel/include/thd.h

```diff
@@ -485,6 +485,16 @@ thd_invstk_current(struct thread *curr_thd, unsigned long *ip, unsigned long *sp
 	return &curr->comp_info;
 }
 
+static inline struct comp_info *
+thd_invstk_current_fault(struct thread *curr_thd, unsigned long *ip, unsigned long *sp, unsigned long *fault_flag, struct cos_cpu_local_info *cos_info)
```

Comment:
I don't see a change to `thd_invstk_current`.  Seems like the correct option is either to 

1. just change the implementation to use the fault parameter, or 
2. change `thd_invstk_current` to call this function.

It seems like your current implementation is *replicating* code with this function.  That is *really bad*.